python/cusignal/filtering/filtering.py

```diff
@@ -183,54 +185,23 @@ def lfilter(b, a, x):
     -------
     y : array
         The output of the digital filter. Output will be clipped to -1 to 1.
-
     """
-    # pack batch
-    shape = x.shape
-    x = x.reshape(-1, shape[-1])
-
-    assert (a.shape[0] == b.shape[0])
```

Comment:
Does assert tell the user to ensure a and b are the same length? Maybe if len a != len b we trailing pad the smaller array to match the length of the end?

Reply:
Okay. I need to investigate how `signal.lfilter` is padding **a** and **b**. Torchaudio does not so luigi and I didn't implement it.

Reply:
A couple things we could do here:

* I *think* if you assert, the user is given a vague AssertionError. Let's add the message that a and b must be the same shape and to zero pad if necessary

* Even more helpful would be to take the max len of a and b and to trailing-zero pad the shorter array if the sizes aren't equal.

Reply:
This is what I currently have.
```python
if a.shape[0] != b.shape[0] or len(a.shape) > 1 or len(b.shape) > 1:
        raise ValueError("Inputs a and b must both be 1D arrays")
```

Reply:
Let me test padding real quick

Reply:
Some code:

```python
import cupy as cp

len_array = max(a.shape[0], b.shape[0])
a = cp.pad(a, (0, len_array - a.shape[0])
b = cp.pad(b, (0, len_array - b.shape[0])
```